### PR TITLE
[GeoMechanicsApplication] Zero mass for geo interface

### LINKED ISSUE #12638: [GeoMechanicsApplication] Let member `CalculateMassMatrix` (of U-Pw interface elements) return the zero matrix
It has been shown that the calculation of the mass matrix for U-Pw continuum elements was incorrect (see [this PR](https://github.com/KratosMultiphysics/Kratos/pull/12521)). However, a similar mistake is present in the calculation of the mass matrix for U-Pw interface elements. @WPK4FEM pointed out that computing the mass matrix for a U-Pw interface element only makes sense if we consider it be filled with water. So far, we don't support such functionality (yet). Therefore, we felt it would be better to remove the incorrect function (i.e. `AssembleDensityMatrix`) entirely, and to return a zero matrix when asking the interface element to compute its mass matrix.

### PR DESCRIPTION
**Pull request by WPK4FEM**
**📝 Description**
Interfaces of infinitesimal thickness have no contribution to inertia.

**🆕 Changelog**
- Removed fill of mass matrix for existing geo interface elements
- Removed helper function to fill directional density matrix.

### CODE REVIEW

File: applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_interface_element.cpp [hunk 1]
```diff
@@ -153,66 +153,11 @@ void UPwSmallStrainInterfaceElement<TDim, TNumNodes>::CalculateMassMatrix(Matrix
 {
     KRATOS_TRY
 
+    // Resizing mass matrix, no mass in these interface elements
     const unsigned int N_DOF = this->GetNumberOfDOF();
-
-    // Resizing mass matrix
-    if (rMassMatrix.size1() != N_DOF) rMassMatrix.resize(N_DOF, N_DOF, false);
+    rMassMatrix.resize(N_DOF, N_DOF, false);
     noalias(rMassMatrix) = ZeroMatrix(N_DOF, N_DOF);
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_interface_element.cpp]: I guess that when we remove the `noalias` construct, we can simplify this further to:
```
    rMassMatrix = ZeroMatrix{this->GetNumberOfDOF(), this->GetNumberOfDOF()};
```

In that case, I would also suggest to remove the lines containing `KRATOS_TRY` and `KRATOS_CATCH`.

**Review by avdg81 — COMMENTED** [review]
I believe this is a nice improvement. I have only one minor suggestion.

File: applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_interface_element.cpp (hunk 1 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_interface_element.cpp]: done ( so I hope ).

**Review by avdg81 — APPROVED** [review]
I'm satisfied with these changes. Let's merge it.

**Pull request merged**